src/cron/fetchExchangeRates.ts

```diff
@@ -0,0 +1,16 @@
+import { getOpenExchangeRates } from '../integrations/openExchangeRates';
+import { logger } from '../logger';
+import type { Cron } from './cron';
+
+export const fetchExchangeRates: Cron = {
+  name: 'fetch-exchange-rates',
+  handler: async () => {
+    logger.info('Fetching exchange rates');
+    try {
+      await getOpenExchangeRates();
```

Comment:
Why do we need the cron, just calling a function should be fine, and then we cache it to redis until expiration?

Reply:
All following calls will be fast.

Reply:
Because we only have 1000 API calls per month, so it'll update the exchange rates once per hour

Reply:
But same effect would be done if we just set expiration in redis to 1 hour, it would be even less calls since if not needed it would not fire as cron would do.

Reply:
Refactored and set TTL to 1 hour and it fetches in the webhook pre-handler if it doesn't exist in redis